src/test/java/tests/ShopTests/ParameterizedUserDetailsTest.java

```diff
@@ -0,0 +1,51 @@
+package tests.ShopTests;
+
+import com.endava.models.Item;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import pages.*;
+import tests.BaseTest;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParameterizedUserDetailsTest extends BaseTest {
+    private LoginPage loginPage;
+    private HomePage homePage;
+    private CartPage cartPage;
+    private UserDetailsPage userDetailsPage;
+
+    @BeforeEach
+    public void setup() {
+        driverSetup();
+        loginPage = new LoginPage(driver);
+        homePage = new HomePage(driver);
+        cartPage = new CartPage(driver);
+        userDetailsPage = new UserDetailsPage(driver);
+        loginPage.openPage();
+        loginPage.userLogin(LoginPage.USERNAME, LoginPage.PASSWORD);
+    }
+
+    @ParameterizedTest(name = "{index}: firstName=''{0}'' lastName=''{1}'' postalCode =''{2}''")
+    @CsvSource({
+            "Krisi, Teneva, '', Error: Postal Code is required",
+            "Krisi, '', 6000, Error: Last Name is required",
+            "'', Teneva, 6000, Error: First Name is required"
+    })
+    void verifyUnsuccessfulCheckoutWithWrongDetails(String firstName, String lastName, String postalCode, String expectedMessage) {
+        List<Item> expectedItems = new ArrayList<>();
```

Comment:
No need to check if items added in cart are the same, you already have different tests for that. If adding an item in the cart is needed as precondition to get to checkout screen, you can put it in the Before method as an action, but without validations - just add random item to cart and proceed to checkout to do the actual validations for error messages etc.